src/game.cpp

```diff
@@ -6736,55 +6940,77 @@ void game::print_visibility_indicator( visibility_type visibility )
     } else if( u.has_effect( effect_blind ) || u.worn_with_flag( "BLIND" ) ) {
-        mvwprintz( w_look, ++line, column, c_yellow,
-                   _( "There's something there, but you can't see what it is." ) );
-        return;
+        foldstring( info, colorize( _( "There's something there, but you can't see what it is." ),
+                                    c_yellow ), max_width );
     } else {
         std::map<std::string, int> item_names;
         for( auto &item : m.i_at( lp ) ) {
             ++item_names[item.tname()];
         }
 
-        const int max_width = getmaxx( w_look ) - column - 1;
+        std::string tmpStr;
         for( const auto &it : item_names ) {
-            if( line >= last_line - 2 ) {
-                mvwprintz( w_look, ++line, column, c_yellow, _( "More items here..." ) );
-                break;
-            }
-
             if( it.second > 1 ) {
-                trim_and_print( w_look, ++line, column, max_width, c_white,
-                                pgettext( "%s is the name of the item. %d is the quantity of that item.", "%s [%d]" ),
-                                it.first.c_str(), it.second );
+                tmpStr = string_format( pgettext( "%s is the name of the item. %d is the quantity of that item.",
+                                                  "%s [%d]" ),
+                                        it.first, it.second );
             } else {
-                trim_and_print( w_look, ++line, column, max_width, c_white, it.first );
+                tmpStr = it.first;
             }
+            info.push_back( trim_to_width( colorize( tmpStr, c_white ), max_width ) );
         }
     }
+
+    return info.size() - old_size;
 }
 
-void game::print_graffiti_info( const tripoint &lp, const catacurses::window &w_look,
-                                const int column, int &line,
-                                const int last_line )
+size_t game::graffiti_info( std::vector<std::string> &info, const tripoint &lp,
+                            const int max_width )
 {
-    if( line > last_line ) {
-        return;
-    }
+    size_t old_size = info.size();
 
-    const int max_width = getmaxx( w_look ) - column - 2;
     if( m.has_graffiti_at( lp ) ) {
-        fold_and_print( w_look, ++line, column, max_width, c_light_gray, _( "Graffiti: %s" ),
-                        m.graffiti_at( lp ).c_str() );
+        std::string graffiti_str = colorize( string_format( _( "Graffiti: %s" ), m.graffiti_at( lp ) ),
+                                             c_light_gray );
+        foldstring( info, graffiti_str, max_width );
     }
+
+    return info.size() - old_size;
+}
+
+size_t game::sound_info( std::vector<std::string> &info, const tripoint &lp, const int max_width )
+{
+    size_t old_size = info.size();
+    auto this_sound = sounds::sound_at( lp );
+
+    if( !this_sound.empty() ) {
+        foldstring( info, string_format( _( "You heard %s from here." ), this_sound ), max_width );
```

Comment:
Player might be deaf or the sound might be too faint to hear, but I guess you're just copying the existing bad code.

Reply:
yes, I am not sure yet of what would be the relevant check to know if he's deaf, but I can add it

Reply:
This code, from `sounds::process_sound_markers`:
```c++
    bool is_deaf = p->is_deaf();
    const float volume_multiplier = p->hearing_ability();
    const int weather_vol = weather_data( g->weather ).sound_attn;

    for( const auto &sound_event_pair : sounds_since_last_turn ) {
        const tripoint &pos = sound_event_pair.first;
        const sound_event &sound = sound_event_pair.second;

        const int distance_to_sound = rl_dist( p->pos().x, p->pos().y, pos.x, pos.y ) +
                                      abs( p->pos().z - pos.z ) * 10;
        const int raw_volume = sound.volume;

        // The felt volume of a sound is not affected by negative multipliers, such as already
        // deafened players or players with sub-par hearing to begin with.
        const int felt_volume = static_cast<int>( raw_volume * std::min( 1.0f,
                                volume_multiplier ) ) - distance_to_sound;

        // Deafening is based on the felt volume, as a player may be too deaf to
        // hear the deafening sound but still suffer additional hearing loss.
        const bool is_sound_deafening = rng( felt_volume / 2, felt_volume ) >= 150;

        // Deaf players hear no sound, but still are at risk of additional hearing loss.
        if( is_deaf ) {
            continue;
        }
        // The heard volume of a sound is the player heard volume, regardless of true volume level.
        const int heard_volume = static_cast<int>( ( raw_volume - weather_vol ) *
                                 volume_multiplier ) - distance_to_sound;

        if( heard_volume <= 0 && pos != p->pos() ) {
            continue;
        }
```